File: tooling/archetypes/camel-archetype-component/src/main/resources/archetype-resources/src/main/java/__name__Component.java
package ${package};

import java.util.Map;

import org.apache.camel.Endpoint;
import org.apache.camel.impl.DefaultComponent;

/**
 * Represents the component that manages {@link ${name}Endpoint}.
 */
public class ${name}Component extends DefaultComponent {

    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
        Endpoint endpoint = new ${name}Endpoint(uri, this);
        setProperties(endpoint, parameters);
        return endpoint;
    }
}

File: tooling/archetypes/camel-archetype-component-scala/src/main/resources/archetype-resources/src/main/scala/__name__Component.scala
package ${package}

import org.apache.camel.Endpoint
import org.apache.camel.impl.DefaultComponent
import java.util.Map

/**
 * Represents the component that manages {@link ${name}Endpoint}.
 */
class ${name}Component extends DefaultComponent {

  protected def createEndpoint(uri: String, remaining: String, parameters: Map[String, Object]): Endpoint = {

    val endpoint = new ${name}Endpoint(uri, this)
    setProperties(endpoint, parameters)

    endpoint
  }
}
